### PR TITLE
feat: add create dividend component

### PR DESCRIPTION
**Pull request by BLasan**
## Description
Create Dividend component added.

## Related issues and discussion
#770 

## Screenshots, if any
Screenshot from 2020-05-31 19-27-33

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

**Comment by BLasan** [comment]
@karantakalkar Please review

**Comment by karantakalkar** [comment]
I'll review by weekend, busy with code rn.

**Comment by BLasan** [comment]
@karantakalkar Done

File: src/app/products/products-routing.module.ts [hunk 1]
```diff
@@ -203,6 +204,14 @@ const routes: Routes = [
                   resolve: {
                     shareProductAndTemplate: ShareProductAndTemplateResolver
                   }
+                },
+                {
+                  path: 'createdividend',
```
Comment (karantakalkar) [inline src/app/products/products-routing.module.ts:209]: Routing is incorrect, see routing of other create forms, also there is dividends list page #804  before initiate dividend form,

File: src/app/products/share-products/view-share-product/view-share-product.component.html [hunk 2]
```diff
@@ -3,7 +3,7 @@
     <fa-icon icon="edit"></fa-icon>&nbsp;&nbsp;
     Edit
   </button>
-  <button mat-raised-button color="primary">
+  <button mat-raised-button color="primary" [routerLink]="['createdividend']">
```
Comment (karantakalkar) [inline src/app/products/share-products/view-share-product/view-share-product.component.html:6]:
```suggestion
  <button mat-raised-button color="primary" [routerLink]="['create']">
```
fix routing, make this a child route of view dividends list page

**Comment by BLasan** [comment]
@karantakalkar I'll refactor now

File: src/app/products/share-products/view-share-product/view-share-product.component.html [hunk 3]
```diff
@@ -3,7 +3,7 @@
     <fa-icon icon="edit"></fa-icon>&nbsp;&nbsp;
     Edit
   </button>
-  <button mat-raised-button color="primary">
+  <button mat-raised-button color="primary" [routerLink]="['create']">
```
Comment (karantakalkar) [inline src/app/products/share-products/view-share-product/view-share-product.component.html:6]:
```suggestion
  <button mat-raised-button color="primary>
```
Will merge this after 804, this is not the right place for the button.

Comment (BLasan) [reply src/app/products/share-products/view-share-product/view-share-product.component.html:6]: ok

**Comment by karantakalkar** [comment]
@BLasan you can sync now

**Comment by BLasan** [comment]
@karantakalkar Ok

File: src/app/products/products-routing.module.ts [hunk 4]
```diff
@@ -70,6 +70,7 @@ import { TaxComponentTemplateResolver } from './manage-tax-components/tax-compon
 import { EditChargeComponent } from './charges/edit-charge/edit-charge.component';
 import { ChargesTemplateResolver } from './charges/charges-template.resolver';
 import { TaxGroupResolver } from './manage-tax-groups/tax-group.resolver';
+import { CreateDividendComponent } from './share-products/create-dividend/create-dividend.component';
```
Comment (Rishabh106) [inline src/app/products/products-routing.module.ts:73]: This should be present under 'Custom Components.'

**Comment by BLasan** [comment]
@karantakalkar chart.js has been removed. Cannot run

**Comment by karantakalkar** [comment]
run npm install

**Comment by BLasan** [comment]
@karantakalkar Please review

**Comment by muskankhedia** [comment]
@BLasan Please resolve the merge conflicts.

**Comment by BLasan** [comment]
> @BLasan Please resolve the merge conflicts.

Done

**Comment by BLasan** [comment]
> Please make this fix

Done. Please review

**Pull request merged**